charon/src/ids/vector.rs

```diff
@@ -300,6 +307,16 @@ where
         ret.elem_count = ret.iter().count();
         ret
     }
+
+    /// Inserts the element at the given index.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `index` > `self.elem_count`.
+    pub fn insert(&mut self, index: I, element: T) {
+        self.vector.insert(index, Some(element));
+        self.elem_count += 1;
```

Comment:
This is almost surely an incorrect API for this type, because inserting not-at-the-end shifts the indices of the other entries, but this type cares very much about indices.